Foundation/NSISO8601DateFormatter.swift

```diff
@@ -10,70 +10,111 @@
-        
+
         public static var withWeekOfYear = ISO8601DateFormatter.Options(rawValue: 1 << 2)
         
-        public static var withDay = ISO8601DateFormatter.Options(rawValue: 1 << 3)
+        public static var withDay = ISO8601DateFormatter.Options(rawValue: 1 << 4)
         
-        public static var withTime = ISO8601DateFormatter.Options(rawValue: 1 << 4)
+        public static var withTime = ISO8601DateFormatter.Options(rawValue: 1 << 5)
         
-        public static var withTimeZone = ISO8601DateFormatter.Options(rawValue: 1 << 5)
+        public static var withTimeZone = ISO8601DateFormatter.Options(rawValue: 1 << 6)
         
-        public static var withSpaceBetweenDateAndTime = ISO8601DateFormatter.Options(rawValue: 1 << 6)
+        public static var withSpaceBetweenDateAndTime = ISO8601DateFormatter.Options(rawValue: 1 << 7)
         
-        public static var withDashSeparatorInDate = ISO8601DateFormatter.Options(rawValue: 1 << 7)
+        public static var withDashSeparatorInDate = ISO8601DateFormatter.Options(rawValue: 1 << 8)
         
-        public static var withColonSeparatorInTime = ISO8601DateFormatter.Options(rawValue: 1 << 8)
+        public static var withColonSeparatorInTime = ISO8601DateFormatter.Options(rawValue: 1 << 9)
         
-        public static var withColonSeparatorInTimeZone = ISO8601DateFormatter.Options(rawValue: 1 << 9)
+        public static var withColonSeparatorInTimeZone = ISO8601DateFormatter.Options(rawValue: 1 << 10)
         
-        public static var withFullDate = ISO8601DateFormatter.Options(rawValue: 1 << 10)
+        public static var withFullDate = ISO8601DateFormatter.Options(rawValue: withYear.rawValue + withMonth.rawValue + withDay.rawValue + withDashSeparatorInDate.rawValue)
         
-        public static var withFullTime = ISO8601DateFormatter.Options(rawValue: 1 << 11)
-        
-        public static var withInternetDateTime = ISO8601DateFormatter.Options(rawValue: 1 << 12)
+        public static var withFullTime = ISO8601DateFormatter.Options(rawValue: withTime.rawValue + withTimeZone.rawValue + withColonSeparatorInTime.rawValue + withColonSeparatorInTimeZone.rawValue)
+
+        public static var withInternetDateTime = ISO8601DateFormatter.Options(rawValue: withFullDate.rawValue + withFullTime.rawValue)
     }
+
 }
 
 open class ISO8601DateFormatter : Formatter, NSSecureCoding {
     
+    typealias CFType = CFDateFormatter
+    private var __cfObject: CFType?
+    private var _cfObject: CFType {
+        guard let obj = __cfObject else {
+            #if os(OSX) || os(iOS)
+                let format = CFISO8601DateFormatOptions(rawValue: formatOptions.rawValue)
+            #else
+                let format = CFISO8601DateFormatOptions(self.formatOptions.rawValue)
+            #endif
+            let obj = CFDateFormatterCreateISO8601Formatter(kCFAllocatorSystemDefault, format)!
```

Comment:
Given the limitations that are being placed on the values that can be passed into the CF function it shouldn't ever be able to return nil but I'm not 100% that is the case so I'll look into it. It may actually just be nicer if we just unwrapped the optional safely anyway rather than just using a force unwrap.